x/ibc/endblock.go

```diff
@@ -1,20 +1,63 @@
 package ibc
 
 import (
+	"fmt"
+
 	sdk "github.com/cosmos/cosmos-sdk/types"
 )
 
 func EndBlocker(ctx sdk.Context, keeper Keeper) {
 	if len(keeper.packageCollector.collectedPackages) == 0 {
 		return
 	}
-	var attributes []sdk.Attribute
+	var (
+		attributes []sdk.Attribute
+		events     sdk.Events
+	)
 	for _, ibcPackageRecord := range keeper.packageCollector.collectedPackages {
 		attributes = append(attributes,
 			sdk.NewAttribute(ibcPackageInfoAttributeKey,
 				buildIBCPackageAttributeValue(ibcPackageRecord.destChainID, ibcPackageRecord.channelID, ibcPackageRecord.sequence)))
 	}
+
 	keeper.packageCollector.collectedPackages = keeper.packageCollector.collectedPackages[:0]
 	event := sdk.NewEvent(ibcEventType, attributes...)
-	ctx.EventManager().EmitEvent(event)
+	events.AppendEvent(event)
+	if sdk.IsUpgrade(sdk.BCFusionThirdHardFork) {
```

Comment:
`IsUpgrade` will be true after the hardfork, and trigger in each block. 

While the `closeSideChainChannels` only need to do once on the `Harfork` block

Reply:
This is not fixed yet

Reply:
fixed

Reply:
you did not do `SetBSCAllChannelClosed`.

Reply:
`SetBSCAllChannelClosed` is invoked multi times now.